### ISSUE TITLE
MinimumIdle affects MaximumPoolSize

### ISSUE THREAD

**Issue by isopov** [brettwooldridge/HikariCP#1258]
AFAIU this is intended behavior that is present for years, but still create this issue so it can be closed as wontfix at least.

 I have many microservices - some with many RPS and some with only a few RPM. They all use the same database with limited number of max connections. One day I faced a problem that some microservice could not start because all connections were exhasted on DB side. I used defaults at that time and MaximumPoolSize of 10 seemed reasonable default to me. But I overriden MinimumIdle to 1 from default 10 for all my microservices to meet DB requirements regarding max connections. And only on prod environment I discovered that this config also set MaximumPoolSize to 1 too.

Here is code sample
```
        try(HikariDataSource ds = new HikariDataSource()){
            ds.setJdbcUrl("jdbc:h2:mem:test_mem");
            ds.setMinimumIdle(1);
            try (Connection ignored = ds.getConnection()) {}
            System.out.println(ds.getMaximumPoolSize());
        }
```
And it prints 1.
Removing `ds.setMinimumIdle(1);` this code prints 10.

TLDR: please remove setting default MaximumPoolSize to MinimumIdle.

**Comment by billoneil** [comment]
The relevant code is [here](https://github.com/brettwooldridge/HikariCP/blob/dev/src/main/java/com/zaxxer/hikari/HikariConfig.java#L999). You just need to set the MaximumPoolSize as well and it will not be overridden.

**Comment by isopov** [comment]
@billoneil Yes, I just wanted to leave a note in issue-tracker that this may seem counterintuitive, while I do understand that it is done to promote fixed-sized pools.

**Comment by douph1** [comment]
Behaviour changed since HikariCP-3.3.1
https://github.com/brettwooldridge/HikariCP/commit/86612b4bc4fb84e9f2f417c22d11dcde84dd369e#diff-10c18d20c1364f48e0b2c8c7ae20f718

https://github.com/brettwooldridge/HikariCP/blob/dev/CHANGES
* do not default maxPoolSize to minIdle when only minIdle is specified

**Comment by isopov** [comment]
@douph1 thanks, than this issue is resolved
